src/text-field.js

```diff
@@ -30,81 +27,48 @@ class Textfield extends React.Component {
   }
 
   render(){
-
     const {
-      className,
-      type,
-      label,
-      floatingLabel,
-      error,
+      children,
       expandable,
-      button,
       icon,
-      expandableHolder
+      floating,
+      className
     } = this.props;
 
-    let {
-      id
-    } = this.props;
-
-    const outerClasses = classnames(outerBaseClasses, {
-      'mdl-textfield--floating-label': floatingLabel,
-      'mdl-textfield--expandable': expandable,
-      'mdl-button': button,
-      'mdl-js-button': button,
-      'mdl-button--icon': icon,
-      'mdl-textfield__expandable-holder': expandableHolder
-    }, className);
-
-    const inputClasses = classnames(inputBaseClasses);
+    let {id} = this.props;
 
-    const labelClasses = classnames(labelBaseClasses);
-
-    let textField;
-    let labelField;
-    let errorField;
-
-    if (!id) {
-      id = '_' + Math.random().toString(36).slice(2);
+    if(!id){
+      id = this._autoId;
     }
 
-    if (type === 'textarea') {
-      textField = (<textarea className={inputClasses} id={id} />);
-    } else {
-      textField = (<input className={inputClasses} type={type} id={id} />);
-    }
-
-    if (label) {
-      labelField = (<label className={labelClasses} htmlFor={id}>{label}</label>);
-    }
+    const inputChildren = React.Children.map(this.props.children, function(child) {
+      return React.cloneElement(child, {
```

Comment:
this is looking good, however, you need to check the type of component each child is.   If the type is TextFieldLabel, you clone with the `htmlFor` property, if it is type TextAreaInput or TextFieldTextarea, you clone with the `id` property.  You also need to check the properties on each child to make sure the `id` or `htmlFor` properties weren't already specified before cloning because we want users to be able to assign their own.